### PR TITLE
Fix bug when reinstalling a candidate (checksums for cached archives)

### PR DESCRIPTION
**Pull request by hgeraldino**
Issue https://github.com/sdkman/sdkman-cli/issues/1005

### CODE REVIEW

File: src/main/bash/sdkman-install.sh [hunk 1]
```diff
@@ -193,6 +194,12 @@ function __sdkman_checksum_zip() {
 	local algorithm checksum cmd
 	local shasum_avail=false
 	local md5sum_avail=false
+	
+	if [ ! -f "${headers_archive}" ]; then
```
Comment (marc0der) [inline src/main/bash/sdkman-install.sh:198]: I must have overlooked this variable name in the past, but it's a bit confusing to me. It isn't pointing to a headers archive but a headers file. Would you mind renaming it for the sake of clarity?

**Review by marc0der — COMMENTED** [review]
This looks good so far, thanks!

One scenario that might have been overlooked is the following:
Many JDKs are packaged as tarballs. So what we do (not to break caching) is repackage them as zips. When installing a previously installed JDK, we go looking for the zip file in the archives folder first.

I haven't verified this, but I suspect that the hash (for the tarball) will be used to check the repacked zip archive and will most probably fail.

The fact that we're repackaging this a as zip file isn't good, and I hope to improve on this in the future, but for now, it is what it is :smile:

File: src/main/bash/sdkman-install.sh (hunk 1 above)
Comment (hgeraldino) [reply src/main/bash/sdkman-install.sh]: s/headers*/headers_file//

**Comment by hgeraldino** [comment]
I think we can handle those cases on the broker side, maybe by not writing the checksum headers for tarballs (so checks aren't performed by the `cli`), adding a `skip-checksum` header or something in that line.

Now I'm curious to know how repackaging these as zip files help with caching 🤔

**Comment by marc0der** [comment]
So I wouldn't recommend we drop the checksums for tarballs, as most JDKs are packaged as such.

The JDKs were the first tarballs we brought onto the platform. Up to that point, they were all zips. When I introduced these tarballs, I didn't feel like rewriting the entire caching mechanism (which was geared for zips), so I decided to repackage the tarballs as zips in the interim. It has served its purpose well until now that we want to do some validation on the cached artefacts.

**Comment by marc0der** [comment]
Also, I should have mentioned that we will revisit the caching strategy (if it's still needed because bandwidth is cheaper than when I initially wrote SDKMAN!) as we rewrite the installation process with Rust.

**Comment by helpermethod** [comment]
IMHO the current caching strategy doesn't make sense anymore. It *very* rarely happens that I reinstall an SDK, and even then my crappy internet connection is good enough to download it on demand.

In the other hand all those cached archives occupied more than 4GB on my hard drive 🥲.

**Comment by hgeraldino** [comment]
Ok, while the caching strategy is decided, this patch will just skip checksums for cached artifacts

**Review by marc0der — APPROVED** [review]
Awesome work, thank you!

**Pull request merged**